mongodb: bump to 4.0.1

grueni opened the pull request:
Please check the addition of the setting of "num_jobs" in patches/04-SConstruct.patch.
This settings defines the number of compile jobs as the number of cores - 1.

Commit:
mongodb: bump to 4.0.1

pyhalov commented on components/database/mongodb-40/patches/03-file-locations.patch:
```diff
@@ -0,0 +1,11 @@
+ If you
+ installed MongoDB using a package management system, check the
+-\fB/etc/mongodb.conf\fP file provided by your packages to see the
++\fB/etc/mongodb/3.4/mongod.conf\fP file provided by your packages to see the
```
Please, fix path

pyhalov commented on components/database/mongodb-40/patches/04-SConstruct.patch:
```diff
@@ -0,0 +1,49 @@
+ import libdeps
+ 
++SetOption('num_jobs', max(1,psutil.cpu_count()-1))
++
```
Can we set it in Makefile?

grueni replied:
Yes, if I would know a method how to put the output of
CORES=´psrinfo -v|grep '^Status of'|tail -1|awk '{print $5}'`           does not work!
into a variable in the Makefile and then use
COMPONENT_BUILD_ARGS += -j $(CORES)
This will fail if the system has only core!

Commit:
mongodb: fix path in patch file

Mno-hime commented on components/database/mongodb-40/Makefile:
```diff
@@ -0,0 +1,129 @@
+#
+# Copyright 2017 Alexander Pyhalov
+# Copyright 2018 Michal Nowak
+#
```
I think your copyright should go here. Same for the rest of the files you edited.

Commit:
mongodb: add copyright, move setting of parallel job ti Makefile

Commit:
mongodb: add services for shards and replsets

Commit:
mongodb:
 * add copyright, move setting of parallel job to Makefile
 * add services for shards and replsets
 * change service name of cluster services
 * used dotted yaml format in config files (better for augeas)

pyhalov commented on components/database/mongodb-40/files/mongodb_40.xml:
```diff
@@ -0,0 +1,55 @@
+-->
+<service_bundle type="manifest" name="mongodb">
+
+ <service name="application/database/mongod_40" type="service" version="1">
```
Should be application/database/mongodb_40

pyhalov commented on components/database/mongodb-40/files/mongodb_cluster_40.xml:
```diff
@@ -0,0 +1,127 @@
+-->
+<service_bundle type="manifest" name="mongodb">
+
+  <service name="application/database/mongos_router_40" type="service" version="1">
```
Why not just application/database/mongos_40 ?

pyhalov commented on components/database/mongodb-40/files/mongodb_cluster_40.xml:
```diff
@@ -0,0 +1,127 @@
+    </property_group>
+
+    <property_group name='options' type='application'>
+       <propval name='config_router' type='astring' value='/etc/mongodb/4.0/mongos_router.conf' />
```
Why not just "config" and mongos.conf?

pyhalov commented on components/database/mongodb-40/files/mongodb_cluster_40.xml:
```diff
@@ -0,0 +1,127 @@
+Copyright 2012 EveryCity Ltd. All rights reserved.
+
+-->
+<service_bundle type="manifest" name="mongodb">
```
I'd avoided delivering whole cluster configuration example. That's fine to deliver just mongos SMF configuration, but cluster configuration can be a) rather specific, b) it's easy to configure it on the base of existing configs for those who need it.

Commit:
mongodb: removed cluster example, added mongos service

grueni commented:
Sorry for the delay. I was very engaged in earning money.
I removed the cluster service and added a mongos service.
I will hold the mongodb-40/cluster in our local repository because we are using it in real life.

pyhalov commented:
While running tests, I see a lot of 'std::exception: locale::facet::_S_create_c_locale name not valid in test ErrorsIfCacheIsEmpty' .

What locale do you use, en_US.UTF-8?

grueni commented:
# locale
LANG=
LC_CTYPE="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_COLLATE="C"
LC_MONETARY="C"
LC_MESSAGES="C"
LC_ALL=
# echo $TZ
Europe/Berlin

2 tests failed
[executor] 2018-09-10T19:09:26.664+0200 Summary: 351 test(s) ran in 419.61 seconds (349 succeeded, 0 were skipped, 2 failed, 0 errored)
    The following tests failed (with exit code):
        build/opt/mongo/client/mongo_uri_test (17)
        build/opt/mongo/util/dns_query_test (17)
I have no connection to the internet from this network and so one test failed. The other is just wrong because the DNS entry does not exist.

pyhalov committed:
mongodb:
- update to 4.0.2
- use env -i while running tests
- revert packaging changes
